apps/settings/js/panels/homescreen_list/update_check.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Homescreen update checking functionality.
+ *
+ * @module UpdateCheck
+ */
+define(() => {
```

Comment:
Let's refactor it as a module to share with `about/update_check.js`. And I suggest you to do it after bug 1161927 (system update API) landed.